frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyMfaCodeHandler.java

```diff
@@ -120,10 +122,13 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                 return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1002);
             }
 
-            var errorResponse = mfaCodeValidator.validateCode(codeRequest.getCode());
+            var errorResponse = mfaCodeValidator.validateCode(codeRequest);
 
-            processCodeSession(
-                    errorResponse, session, mfaMethodType, input, userContext, isRegistration);
+            if (errorResponse.filter(t -> t.equals(ErrorResponse.ERROR_1041)).isPresent()) {
```

Comment:
Small suggestion to use a method reference rather than a lambda
```suggestion
            if (errorResponse.filter(ErrorResponse.ERROR_1041::equals).isPresent()) {
```

Reply:
Good shout 👍  I'll update this.

Reply:
Done